backend/fms_core/resources/experiment_run.py

```diff
@@ -0,0 +1,279 @@
+
+__all__ = ["ExperimentRunResource"]
+
+
+process_content_type = ContentType.objects.get(app_label="fms_core", model="process")
+
+
+class ExperimentRunResource(GenericResource):
+    experiment_id = Field(column_name='Experiment ID')
+    container_barcode = Field(column_name='Experiment Container Barcode',
+                              widget=ForeignKeyWidget(Container, field='barcode'))
+    container_kind = Field(column_name='Experiment Container Kind')
+    instrument_name = Field(column_name='Instrument Name',
+                              widget=ForeignKeyWidget(Instrument, field='name'))
+    start_date = Field(attribute='start_date', column_name='Experiment Start Date', widget=DateWidget())
+
+
+    ERROR_CUTOFF = 500
+
+
+    class Meta:
+        model = ExperimentRun
+        import_id_fields = ()
+        fields = (
+        )
+        excluded = (
+            'container',
+            'instrument',
+            'process',
+        )
+
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        self.sample_rows = []
+        self.experiments = {}
+        self.temporary_experiment_id = None
+        self.transfer_process_by_experiment_id = {}
+
+        self.experiment_type_name = dataset[1][2]
+
+        # at index 6 is the first process name
+        self.protocols_starting_idx = 6
+
+        self.properties_row = list(filter(None, dataset[9][self.protocols_starting_idx:]))
+
+        self.protocols_ending_idx = self.protocols_starting_idx + len(self.properties_row)
+
+
+        # Preload PropertyType objects for this experiment type in a dictionary for faster access
+        self.property_types_by_name = {}
+        for property_column in self.properties_row:
+            self.property_types_by_name[property_column] = PropertyType.objects.get(name=property_column)
+
+        # Preload Protocols objects for this  experiment type in a dictionary for faster access
+        self.protocols_dict = {}
+        protocols_for_experiment_type = PROTOCOLS_BY_EXPERIMENT_TYPE_NAME[self.experiment_type_name]
+        for protocol_name in protocols_for_experiment_type.keys():
+            p = Protocol.objects.get(name=protocol_name)
+            subprotocol_names = protocols_for_experiment_type[protocol_name]
+            self.protocols_dict[p] = map(lambda x: Protocol.objects.get(name=x), subprotocol_names)
+
+
+        skip_rows(dataset, 10)  # Skip preamble
+
+
+    def before_import_row(self, row, **kwargs):
+        self.temporary_experiment_id = row.get("Experiment ID")
+
+        super().before_import_row(row, **kwargs)
+
+
+
+    def import_row(self, row, instance_loader, using_transactions=True, dry_run=False, raise_errors=False, **kwargs):
+        import_result = super().import_row(row, instance_loader, using_transactions, dry_run, raise_errors, **kwargs)
+
+        row_id = row.get("#")
+
+        # We are in the Sample section
+        if isinstance(row_id, int) and row_id > 100:
```

Comment:
This is a bit risky if the users decide to cut unused lines and renumber them...

Reply:
I agree, but in this method you only get an OrderedDict with the content of row. You can situate it within the dataset, you don't even have access to the whole dataset (see image for an example) Do you have a suggestion? 

![image](https://user-images.githubusercontent.com/1617741/125516934-c5f429ea-b76b-4fc5-97cd-749817f0e78f.png)